src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -49,7 +50,13 @@ export default class StudentApplication extends Component {
       applicationID: -1, // Default value of -1 indicate to backend that the application ID number is not yet known
       primaryUsername: null, // The username of the primary applicant
       applicants: [],
+      preferredHalls: [{ hallName: '', hallRank: 1 }],
     };
+    // Off-campus program info is stored as a Map, where the Key is a student's username and the corresponding Value is the department of that student's off-campus program
+    this.offCampusProgramInfo = new Map();
+    this.state.applicants.forEach((profile) =>
+      this.offCampusProgramInfo.set(profile.AD_Username, ''),
+    );
```

Comment:
For this map, the keys are the usernames of each applicant and the values are the department of that student's off-campus program, or empty string '' if not applicable. This may be changed to an array of Javascript custom objects or another data structure in the future when the off-campus programs feature is implemented